mmengine/infer/infer.py

```diff
@@ -0,0 +1,468 @@
+    - postprocess: Postprocess the results returned by :meth:`forward` and
+      :meth:`visualzie`.
+
+    Args:
+        model (str, optional): Path to the config file or the model name
+            defined in metafile. Take the `mmdet metafile <https://github.com/open-mmlab/mmdetection/blob/master/configs/retinanet/metafile.yml>`_
+            as an example, the `model` could be `retinanet_r18_fpn_1x_coco` or
+            its alias.
+        weights: Path to the checkpoint. If it is not specified and model is a
+            model name of metafile, the weights will be loaded from metafile.
+            Defaults to None
+        device (str, optional): Device to run inference. If None, the best
+            device will be automatically used. Defaults to None.
+
+    Note:
+        Since `Inferencer` could be used to inference batch data,
+        `collate_fn` should be defined. If `collate_fn` is not defined in config
+        file, the `collate_fn` will be `pseudo_collate` by default.
+    """  # noqa: E501
+
+    preprocess_kwargs: set = set()
+    forward_kwargs: set = set()
+    visualize_kwargs: set = set()
+    postprocess_kwargs: set = set()
+
+    def __init__(
+        self,
+        model: Union[ModelType, str],
+        weights: Optional[str] = None,
+        device: Optional[str] = None,
+    ) -> None:
+        # Load config to cfg
+        cfg: ConfigType
+        if isinstance(model, str):
+            if osp.isfile(model):
+                cfg = Config.fromfile(model)
+            else:
+                # Load config and weights from metafile. If `weights` is
+                # assigned, the weights defined in metafile will be ignored.
+                cfg, _weights = self._load_model_from_metafile(model)
+                if weights is not None:
+                    weights = _weights
+        elif isinstance(model, (Config, ConfigDict)):
+            cfg = copy.deepcopy(model)
+        elif isinstance(model, dict):
+            cfg = copy.deepcopy(ConfigDict(model))
+        else:
+            raise TypeError('config must be a filename or any ConfigType'
+                            f'object, but got {type(model)}')
+
+        if cfg.model.get('pretrained') is not None:
+            del cfg.model.pretrained
+
+        if device is None:
+            device = get_device()
+
+        self.cfg = cfg
+        self.model = self._init_model(cfg, weights, device)  # type: ignore
+        self.pipeline = self._init_pipeline(cfg)
+        self.collate_fn = self._init_collate(cfg)
+        self.visualizer = self._init_visualizer(cfg)
+
+    def __call__(
+        self,
+        inputs: InputsType,
+        return_datasamples: bool = False,
+        batch_size: int = 1,
+        **kwargs,
+    ) -> dict:
+        """Call the inferencer.
+
+        Args:
+            inputs (InputsType): Inputs for the inferencer.
+            return_datasamples (bool): Whether to return results as
+                datasamples. Defaults to False.
```

Comment:
Suggest adding a link to BaseDataElement documentation.

Reply:
Return the api docs